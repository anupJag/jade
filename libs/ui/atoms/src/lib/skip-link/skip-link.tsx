import React from 'react';
import * as S from './skip-link.styled';

export const SkipLink = () => (
  <S.SkipArea>
    <S.SkipAnchor href="#main-content">skip to main content</S.SkipAnchor>
  </S.SkipArea>
);
