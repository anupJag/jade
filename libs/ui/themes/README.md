# fe-themes

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `ng test fe-themes` to execute the unit tests via [Jest](https://jestjs.io).
