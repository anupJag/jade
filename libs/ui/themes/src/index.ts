export * from './lib/defaultTheme';
export * from './lib/styled';
export * from './lib/themeProvider';
export { default as styled } from './lib/styled';
export * from './lib/GlobalStyles';
