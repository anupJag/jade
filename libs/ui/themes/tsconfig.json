{
  "extends": "../../../tsconfig.json",
  "compilerOptions": {
    "jsx": "react",
    "allowJs": false,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "types": ["node", "jest"]
  },
  "include": ["**/*.js", "**/*.jsx", "**/*.ts", "**/*.tsx"]
}
