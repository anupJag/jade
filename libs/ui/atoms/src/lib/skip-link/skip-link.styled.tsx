import { styled } from '@jade/ui/themes';
import { accessibility } from '@jade/ui/utils';

export const SkipArea = styled.div`
  background: ${props => props.theme.themeColors.background.highlightArea};
`;
export const SkipAnchor = styled.a`
  ${accessibility.srOnly()}
  ${accessibility.srOnlyFocusable()}
  &:focus, &:active {
    color: ${props => props.theme.themeColors.textColor.highlightArea};
    text-decoration: underline;
    padding: ${props => props.theme.space.xs};
    display: inline-block;
    margin: ${props => props.theme.space.xxs};
    ${props => props.theme.themeBorders.highlightedLinksOutline};
    outline-offset: 2px;
  }
`;
