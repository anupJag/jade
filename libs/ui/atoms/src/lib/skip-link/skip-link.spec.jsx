import React from 'react';
import { render } from '@testing-library/react';

import { ThemeProvider } from '@jade/ui/themes';
import { SkipLink } from './skip-link';

describe('<SkipLink />', () => {
  let skipLinkComponent;
  beforeEach(() => {
    skipLinkComponent = render(
      <ThemeProvider>
        <SkipLink />
      </ThemeProvider>,
    );
  });

  /**
   * Capturing Sanpshot : Skip Link
   *
   *  To check style / structure changes in Styled Components
   */
  test('snapshot check', () => {
    const { asFragment } = skipLinkComponent;
    expect(asFragment()).toMatchSnapshot();
  });
});
