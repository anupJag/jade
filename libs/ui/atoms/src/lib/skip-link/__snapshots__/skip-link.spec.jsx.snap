// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SkipLink /> snapshot check 1`] = `
<DocumentFragment>
  <div
    class="css-1djjrov"
  >
    <a
      class="css-81hfzx"
      href="#main-content"
    >
      skip to main content
    </a>
  </div>
</DocumentFragment>
`;
